static/js/containers/pages/CreateCouponPage.js

```diff
@@ -41,74 +40,55 @@ export class CreateCouponPage extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props)
     this.state = {
-      isPromo:          false,
-      productType:      "courserun",
-      selectedProducts: [],
-      created:          false
+      couponId: null
     }
   }
-  onSubmit = async (coupon: Object, { setSubmitting, setErrors }: Object) => {
+  onSubmit = async (
+    couponData: Object,
+    { setSubmitting, setErrors }: Object
+  ) => {
     const { createCoupon } = this.props
-    coupon.product_ids = coupon.products.map(product => product.id)
-    if (coupon.coupon_type === "single-use") {
-      coupon.max_redemptions = coupon.num_coupon_codes
+    couponData.product_ids = couponData.products.map(product => product.id)
+    if (couponData.coupon_type === COUPON_TYPE_SINGLE_USE) {
+      couponData.max_redemptions = couponData.num_coupon_codes
     } else {
-      coupon.num_coupon_codes = 1
+      couponData.num_coupon_codes = 1
     }
-    coupon.amount = coupon.amount / 100
+    couponData.amount = couponData.discount / 100
     try {
-      const result = await createCoupon(coupon)
-      if (result && result.transformed) {
-        this.setState({ created: true })
+      const result = await createCoupon(couponData)
+      if (result.body && result.body.id) {
+        this.setState({ couponId: result.body.id })
       } else if (result.body && result.body.errors) {
         setErrors(_.merge({}, ...result.body.errors))
-        this.setState({ created: false })
+        this.clearSuccess()
```

Comment:
I may be wrong, but I don't think we need to do this anymore since it's either the default state or resets when creating a second coupon.

Reply:
✔️